File: Functions/updated_finance_calc.py
print("This calculator will calculate your budget for the month.")
#getting user inputs

def iput(type):
    var = input(f"How much is your {type}?: ")
    #print(f"Your {type} is $.")
    return f"{var}"

def percent(type, amount):
    per = amount/income*100
    print(f"Your {type} is {per}% of your income.")

income = float(iput("income"))
rent = float(iput("rent"))
utilites = float(iput("utilites"))
groceries = float(iput("groceries"))
transportation = float(iput("transportation"))

#calculations
expenses = rent + utilites + groceries + transportation;
savings = income * 0.2;
total = income - expenses - savings;

#prints
print("Your income is $", income)
print("Your expenses is $", expenses)
print("Your savings is $", savings)
print("Your total is $", total)
percent("rent", rent)
percent("utilites", utilites)
percent("groceries", groceries)
percent("transportation", transportation)

File: Functions/updated_finance_calc.c
#include <stdio.h>
float income, rent, utilites, groceries, trans, expenses, savings, total;

float input(char type[], float var){
    printf("How much is your %s?: \n", type);
    scanf("%f", &var);

    return var;
}

void percent(char type[], int amount){
    int per = amount/income*100;
    printf("\nYour %s cost is %d%%", type, per);
    printf(" of your income.");
}

int main(void){
    printf("This is going to calculate your budget for this month.\n");
    income = input("income", income);
    rent = input("rent", rent);
    utilites = input("utilites", utilites);
    groceries = input("groceries", groceries);
    trans= input("transportation", trans);

    //calculations
    expenses = rent + utilites + groceries + trans;
    savings = income*.2;
    total = income - expenses - savings;

    printf("Your income is: $ %2.f \n", income);
    printf("Your expenses are: $ %.2f \n", expenses);
    printf("Your savings are: $ %.2f \n", savings);
    printf("Your total left to spend is: $ %.2f", total);

    percent("rent", rent);
    percent("utilites", utilites);
    percent("groceries", groceries);
    percent("transportation", trans);
    percent("savings", savings);
    percent("expenses", expenses);

    return 0;
}